### PR TITLE
Enable nullable: System.Management.Automation.Language.ISupportsTypeCaching

### PR DESCRIPTION
**Pull request by xtqqczze**
_Tracking issue: #12631._

### CODE REVIEW

File: src/System.Management.Automation/engine/parser/ast.cs [hunk 1]
```diff
@@ -8406,7 +8408,8 @@ public override int GetHashCode()
         /// </remarks>
         internal bool IsType(Type type)
         {
-            string fullTypeName = type.FullName;
+            string? fullTypeName = type.FullName;
+            Debug.Assert(fullTypeName is not null);
```
Comment (xtqqczze) [inline src/System.Management.Automation/engine/parser/ast.cs:8412]: We probably should add a null check, but I left the behaviour unchanged for this PR.

**Comment by iSazonov** [comment]
@xtqqczze We will move faster if we follow our plan - one type per PR without fixing usages of the type.

**Comment by iSazonov** [comment]
ITypeName is in another PR - please remove.

**Comment by xtqqczze** [comment]
rebased to restrict changes to `ISupportsTypeCaching`

**Review by iSazonov — APPROVED** [review]

**Comment by xtqqczze** [comment]
@powercode Can you review?

**Review by powercode — APPROVED** [review]

**Review by powercode — COMMENTED** [review]
LGTM

**Pull request merged**